mailbox/event/event-rabbitmq/src/test/java/org/apache/james/mailbox/events/RabbitMQEventBusTest.java

```diff
@@ -181,6 +184,35 @@ void registerGroupShouldCreateRetryExchange() throws Exception {
     class ConcurrentTest implements EventBusConcurrentTestContract.MultiEventBusConcurrentContract,
         EventBusConcurrentTestContract.SingleEventBusConcurrentContract {
 
+        @Test
+        void rabbitMQEventBusCannotHandleHugeDispatchingOperations() throws Exception {
+            EventBusTestFixture.MailboxListenerCountingSuccessfulExecution countingListener1 = newCountingListener();
+
+            eventBus().register(countingListener1, new EventBusTestFixture.GroupA());
+            int totalGlobalRegistrations = 1; // GroupA + GroupB + GroupC
+
+            int threadCount = 10;
+            int operationCount = 10000;
+            int totalDispatchOperations = threadCount * operationCount;
+            eventBus = (RabbitMQEventBus) eventBus();
+            ConcurrentTestRunner.builder()
+                .operation((threadNumber, operationNumber) -> eventBus.dispatch(EVENT, NO_KEYS))
+                .threadCount(threadCount)
+                .operationCount(operationCount)
+                .runSuccessfullyWithin(Duration.ofMinutes(10));
+
+            // there is a moment when RabbitMQ EventBus consumed amount of messages, then it will stop to consume more
+            await()
+                .pollInterval(com.jayway.awaitility.Duration.FIVE_SECONDS)
+                .timeout(com.jayway.awaitility.Duration.TEN_MINUTES).until(() -> {
+                    int totalEventsReceived = totalEventsReceived(ImmutableList.of(countingListener1));
```

Comment:
We only have one listener. Calling `countingListener.numberOfEventCalls()` should be enough...

IMO this could very well be inlined in the `assertThat(...)` call.

Reply:
fixed